gpt_json/tests/test_truncation.py

```diff
@@ -0,0 +1,135 @@
+import random
+
+import pytest
+from pydantic import BaseModel
+
+from gpt_json.gpt import GPTJSON
+from gpt_json.models import (
+    GPTMessage,
+    GPTMessageRole,
+    TruncationOptions,
+    VariableTruncationMode,
+)
+from gpt_json.truncation import truncate_tokens
+
+
+def test_fill_messages_truncated():
+    class TestSchema(BaseModel):
+        summary: str
+
+    gpt = GPTJSON[TestSchema](None)
+    assert gpt.fill_messages(
+        [
+            GPTMessage(role=GPTMessageRole.SYSTEM, content="system"),
+            GPTMessage(
+                role=GPTMessageRole.USER,
+                content="some long text: {long_text}",
+            ),
+        ],
+        dict(
+            long_text="hello world goodbye world",
+        ),
+        truncation_options=TruncationOptions(
+            target_variable="long_text",
+            max_prompt_tokens=20,
+            truncation_mode=VariableTruncationMode.BEGINNING,
+        ),
+        max_response_tokens=None,
+    ) == [
+        GPTMessage(role=GPTMessageRole.SYSTEM, content="system"),
+        GPTMessage(role=GPTMessageRole.USER, content="some long text: hello world"),
+    ]
+
+
+def test_fill_messages_truncated_failure_case():
+    class TestSchema(BaseModel):
+        summary: str
+
+    gpt = GPTJSON[TestSchema](None)
+
+    # this should fail because the max_prompt_tokens is too small
+    with pytest.raises(ValueError):
```

Comment:
nit: Instead of a blanket value error, can you check against a regex that validates max_prompt_tokens is the issue here?